std-bits/base/src/main/java/org/enso/base/enso_cloud/HTTPBuilder.java

```diff
@@ -0,0 +1,92 @@
+package org.enso.base.enso_cloud;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/***
+ * Makes HTTP requests with secrets in either header or query string.
+ */
+public class HTTPBuilder {
+  private static String makeQueryAry(EnsoKeyValuePair pair) {
+    String resolvedValue = (pair.secretId() != null && !pair.secretId().isBlank())
+        ? EnsoSecretReader.readSecret(pair.secretId())
+        : pair.value();
+
+    String resolvedKey = pair.key() != null && !pair.key().isBlank() ? URLEncoder.encode(pair.key(), StandardCharsets.UTF_8) + "=" : "";
+    return resolvedKey + URLEncoder.encode(resolvedValue, StandardCharsets.UTF_8);
+  }
+
+  public static Connection getJDBCConnection(String url, EnsoKeyValuePair[] properties)
+    throws SQLException {
+    var javaProperties = new Properties();
+    for (EnsoKeyValuePair(
+        String key, String value, String secretId
+    ) : properties) {
+      var resolvedValue = (secretId != null && !secretId.isBlank())
+          ? EnsoSecretReader.readSecret(secretId)
+          : value;
```

Comment:
This is used in a few places, I think we should have a common method for it.

Maybe
```java
public static String resolveValue(EnsoKeyValuePair pair) { ... }
```
as a static method in `EnsoKeyValuePair`?

I guess it should not be a member method, to not make it too easy to extract the Secret if the `EnsoKeyValuePair` instance leaks to Enso.

It could even be a static method in a private `EnsoKeyValuePairHelper`.

Just IMO would be good to not have this logic duplicated across multiple places (currently I found 3).

Reply:
I just realise the function does not necessarily have to be static and hidden, it will be enough to make it package-private instead of public - then Enso code would not be able to call it anyway.